velox/type/parser/ParserUtil.cpp

```diff
@@ -65,4 +65,37 @@ std::pair<std::string, std::shared_ptr<const Type>> inferTypeWithSpaces(
       fieldName, typeFromString(allWords.substr(fieldName.size() + 1)));
 }
 
+std::unordered_map<std::string, int64_t> parseMapFromString(
+    const std::string& input) {
+  folly::dynamic obj = folly::parseJson(input);
+  VELOX_CHECK(obj.isObject(), "Parsed JSON is not an object");
+  std::unordered_map<std::string, int64_t> result;
+
+  for (const auto& item : obj.items()) {
+    std::string key = item.first.asString();
+    VELOX_CHECK(
+        item.second.isInt(),
+        "Invalid value {} for key {} in enum type",
+        item.second,
+        key);
+    int64_t value = item.second.asInt();
+    result.emplace(std::move(key), value);
```

Comment:
Add a check for duplicates.

Reply:
Thanks for the note. I noticed that folly::parseJson will silently override duplicate keys when parsing JSON maps. So instead, I formatted the map as an array of pairs from the bison parser file so that the same folly::parseJson can be used to also detect and throw on duplicates. Please let me know if manually parsing would look cleaner than manipulating the input to use folly::parseJson. I have left comments explaining this and tests to check for duplicates

Reply:
Thanks, but I don't see these changes.